internal/provider/resource_apply_module.go

```diff
@@ -300,13 +300,14 @@ func (t *applyModuleResource) Read(ctx context.Context,
 		return
 	}
 
-	// Update the state with the computed module source and version.
-	if currentApplied.moduleSource != nil {
+	// If there is no current state version, currentApplied can be nil.
+	// If available/possible, update the state with the computed module source and version.
+	if (currentApplied != nil) && (currentApplied.moduleSource) != nil {
 		state.ModuleSource = types.StringValue(*currentApplied.moduleSource)
 	} else {
 		state.ModuleSource = types.StringNull()
 	}
-	if currentApplied.moduleVersion != nil {
+	if (currentApplied != nil) && (currentApplied.moduleVersion) != nil {
```

Comment:
Parenthesis are not needed around `(currentApplied.moduleSource)` (same for below)